src/Raven.Client/Documents/Conventions/DocumentConventions.cs

```diff
@@ -385,15 +385,29 @@ public int LoadBalancerContextSeed
 
         public LoadBalanceBehavior LoadBalanceBehavior
         {
-            // We have to make this check so if admin activated this, but client code did not provide the selector,
```

Comment:
@ayende I do not think this is correct (the comment) ? IIRC we will use session counter as the key then?

Reply:
No, this is correct. The session key is used only if the `LoadBalanceBehavior.UseSessionContext` is set.
That was done _explicitly_ to only opt in from the client.

Otherwise, each session will use a different node and you won't have the usual mechanics of one primary normally.

Reply:
This is exactly what I would expect after setting Load Balance Behavior to UseSessionContext. When context is not set (either via Selector or Manually) then each session should just use the counter. This is also a form of Load Balancing.

Reply:
That will lead to strange errors. A write goes to one node and a read to the other.
That leads to support calls :-)

I would rather this be very explicit. An application must _choose_ to enable this.

Reply:
And an admin setting is not enough, you may have multiple applications running.

Reply:
What do you mean write goes to one and read goes to other? It will stick to the same node during the session?

Why you want to be so explicit? What is the difference between this and ReadBalanceBehavior? If we want to be so explicit, then I do not see a reason of having LoadBalanceBehavior in the Client Configuration.

Reply:
Separate requests. PRG is a good example.
Two sessions, but same user seeing that, and data goes poof

Reply:
So the only usage in the ClientConfiguration for this is to turn it off? Do we want that or just Seed is enough?

Reply:
Yes